src/hivpy/sexual_behaviour.py

```diff
@@ -100,8 +103,8 @@ def update_sex_groups(self, population: pd.DataFrame):
                     Pmin = np.zeros(subpop_size)
                     Pmax = np.zeros(subpop_size)
                     for new_group in range(dim):
-                        Pmin = Pmax
-                        Pmax += self.prob_transition(sex, ages, prev_group.value, new_group)
+                        Pmin = Pmax.copy()
```

Comment:
Good catch. We'll need to be aware of this, as well as checking whether something returns a view into a dataframe or a copy.